core/cc/archive.cpp

```diff
@@ -29,6 +29,15 @@
 #include <unistd.h>
 #endif
 
+#if defined GAPID_ARCHIVE_USE_MMAP
```

Comment:
We use `#ifdef` elsewhere in the codebase, but I personally prefer to use just `#if` and always define it to `0` or `1`. This way you can get the compiler to warn if the define is missing instead of just enabled / disabled (that said, I'm not sure if we have those compiler flags enabled)